payloads/src/lib.rs

```diff
@@ -1,4 +1,314 @@
 #![no_std]
 
 pub mod external;
-pub mod payload;
+// TODO: serialize doesn't even compile yet
+// mod serialize;
+pub mod payload {
```

Comment:
Sorry for being nitpicky, but I would actually keep it in separate file (e.g. payloads/src/payload.rs).

It would also be great to perform git move to preserve the history of the file:

```
git mv payloads/src/payload/mod.rs payloads/src/payload.rs
```